File: AlgoritmosOrdenamiento/Comparaciones/src/InsertionSort.py
from io import open
from time import time

def insertionSort(arr): 
    for i in range(1, len(arr)): 
        key = arr[i] 
        j = i-1
        while j >= 0 and key < arr[j] : 
                arr[j + 1] = arr[j] 
                j -= 1
        arr[j + 1] = key 

tam = [100000, 300000, 500000, 700000, 1000000]
times = [0,0,0,0,0]
f=open("Salidas/InsertionSort/salida1py.txt","w")

for j in range(5):            
    file = open("Entradas/aleatorio1.txt","r")
    A = file.readlines()
    file.close()

    for i in range(tam[j]):
        A[i]=int(A[i])

    t0=time()
    insertionSort(A) 
    tiempo = time()-t0
    tiempo = round(tiempo,3)

    A.clear()
    f.write(str(tam[j])+" "+str(tiempo)+"\n")

File: AlgoritmosOrdenamiento/Comparaciones/src/InsertionSort.cpp
#include <iostream>
#include <sstream> 
#include <fstream>
#include <ctime> 
#include <string> 

using namespace std;

unsigned t0, t1;

ofstream fs("Salidas/InsertionSort/salida2cc.txt");

template < typename Type >
void InsertionSort( Type *arr, int size ) { 

    for ( int i = 1, j, k; i < size; i++ ) { 
        k = arr[ i ]; 
        j = i - 1; 

        while ( j >= 0 && arr[ j ] > k ) { 
            arr[ j + 1 ] = arr[ j ]; 
            j--; 
        } 
        arr[ j + 1 ] = k; 
    } 
}

int main(){
	int tam [] = {100000, 300000, 500000, 700000, 1000000}; 

    for(int j=0;j<5;j++){
        int *A = new int[tam[j]];

        ifstream file;
        string num;
        int num_int=0;
        file.open("Entradas/aleatorio2.txt",ios::in);
        int i=0;
        while(!file.eof() && i<=tam[j]){
            getline(file,num);
            stringstream convert(num);
            convert>>num_int;
            A[i]=num_int;
            i++;
        }
        file.close();

        t0=clock();
        InsertionSort(A,tam[j]);
        t1 = clock();

        double time = (double(t1-t0)/CLOCKS_PER_SEC);
        fs << tam[j] << " " << time <<endl;
        delete[] A;
    }

	return 0;

} 